src/__experimental__/components/form/form.component.js

```diff
@@ -208,41 +199,66 @@ class FormWithoutValidations extends React.Component {
   triggerSubmit(ev, valid) {
     if (this.props.onSubmit) {
       this.props.onSubmit(ev, valid, this.enableForm);
+    } else if (this.state.formInputs) {
+      this.addOtherInputsToState();
     } else {
       this._form.submit();
     }
   }
 
-  /**
-   * enables a form which has been disabled after being submitted.
-   */
+  /* Add all inputs to state so that all are submitted. */
+  addOtherInputsToState() {
+    Object.keys(this._form.elements).forEach((id) => {
+      const { name, value, type } = this._form.elements[id];
+      const inputName = type === 'hidden' && id === '0' ? 'csrf-token' : name;
```

Comment:
could you extract that conditional to a separate const? Maybe:
`const isCsrfToken = (type === 'hidden' && id === '0');`

Reply:
Agreed will make the changes

Reply:
Addressed in https://github.com/Sage/carbon/pull/2175/commits/27e1fca2dea6feee76c60bbeda6e239da71a4f5a